### ISSUE TITLE
[JetMenu] Fix menu responsiveness during screen resize

### ISSUE THREAD

**Issue by XVII** [CrocoBlock/suggestions#1744]
Currently the menu either goes into mobile or desktop mode on load. Resizing the window (or rotating a device) means the menu won't re-adjust if it goes past the break point.

Calling it responsive in it's current state is a bit misleading:

![2020-08-18-15-42-03](https://user-images.githubusercontent.com/446329/90475085-1c8ce480-e16a-11ea-99b9-be828896b963.gif)

Any other truely responsive menu would automatically switch between mobile and desktop based on the specified breakpoints.

# Incompatible with caching plugins
The current approach uses a is_phone() function which, based on the UA, sets if the menu should in mobile or desktop view. 

However this approach breaks any output caching solutions as a desktop user might cache the page with one menu, and mobile user with another.

If the menu was properly responsive this caching issue would be resolved.

**Comment by mbaierl** [comment]
Yeah, just noted the same. This is not "responsive" and does not work as expected. I filed a bug report with the support team.

**Comment by XVII** [comment]
> Yeah, just noted the same. This is not "responsive" and does not work as expected. I filed a bug report with the support team.

Let me know if you hear anything constructive. Have never had much luck with support aside from "raise a GitHub issue" 🙄

**Comment by mbaierl** [comment]
> Let me know if you hear anything constructive. Have never had much luck with support aside from "raise a GitHub issue" 🙄

Me neither. Nor is anything useful happening here. 
But still - we need to open issues and bug reports to raise awareness.

**Comment by XVII** [comment]
Related to #2252

**Comment by XVII** [comment]
This actually used to happen in V1 of JetMenu but was broken in V2.

@Floydtm , any chance we could get an update here?

**Comment by Floydtm** [comment]
for now I can say what we are thinking about how to fix it.

**Comment by XVII** [comment]
Thanks @Floydtm . It's a big gotcha and the menu was why I originally bought the plugin 🙈

Hoping we see a fix soon.

Probably should remove "responsive" on the plugin's page. Misleading!

**Comment by bluantinoo** [comment]
+1

**Comment by pixelmultiplo** [comment]
Terrible

**Comment by Nicoduweb** [comment]
I can't believe JetMenu is not responsive in 2021. Serioulsy ....

**Comment by KingGrizzly** [comment]
+1 Needs to match the way Elementor behaves with breakpoints: Desktop, Tablet, Mobile.

**Comment by kamyabnazari** [comment]
+1 this is a bit sad. It is now June and still nothing.

**Comment by srack835** [comment]
+1 Breakpoints are incredibly important for menu functionality!

**Comment by Androidiawebsites** [comment]
it's 2022 and so far no solution

**Comment by GALASIO** [comment]
+1

**Comment by Androidiawebsites** [comment]
What???

**Comment by erok415** [comment]
An update to the mega-menu seems long overdue since many people resize their screens to fit 50/50, whether on a desktop or tablet. Refreshing is an outdated solution to showing the mobile menu since this could easily be handled with CSS to reveal the mobile menu. In contrast, CSS is the best-preferred method to change the look and feel of the mobile menu based on media queries. If you want more B2B businesses to use this, they will expect a more professional solution that works on resizing without asking the end user to refresh. This is bad UX. Other than that, the feature works as expected. It's a shame we'll need to use a different method to create a mega-menu and toss this plugin into the trash bin.

**Comment by StenderIternum** [comment]
Almost 3 years later... No Improvement. 
You even need to emulate devices to see the mobile menu - and when the screen is resized to a size where the menu item containing a template, is hidden in your ellipses solution, the template is not loaded. 
making the menu item useless depending on the screen size.

**Comment by XVII** [comment]
Elementor Pro just released Mega Menus in preview. I'll be likely moving to that once it's stable as these plugins aren't well maintained.

**Comment by erok415** [comment]
That’s great news

**Comment by mbaierl** [comment]
Unfortunately the Elementor Mega Menu sucks on mobile... are there any other alternatives?

**Comment by aalexandr21** [comment]
The issue with responsiveness during screen resize was fixed 
https://i.imgur.com/pxgGVGo.png https://i.imgur.com/jB4tPJM.png
